mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/MessageIdMapperTest.java

```diff
@@ -240,6 +240,66 @@ public void deleteShouldDeleteMessageIndicesWhenStoredTwoTimesInTheSameMailbox()
         assertThat(mailboxes).isEmpty();
     }
 
+    @ContractTest
```

Comment:
Aren't we supposed to delete the message when there's no more references to it ?